src/App.tsx

```diff
@@ -1,61 +1,159 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser, User } from './react-app-env';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (userId: number): User | null => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const todoWithUser: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
+  const [todos, setTodos] = useState<TodoWithUser[]>(todoWithUser);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+
+  const addTodo = () => {
+    const maxId = Math.max(...todos.map(todo => todo.id));
+    const newTodo = {
+      id: maxId + 1,
+      title,
+      completed: false,
+      userId: selectedUserId,
+      user: usersFromServer.find(user => user.id === selectedUserId) || null,
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const clearButtonClick = () => {
+    setIsButtonClicked(false);
+  };
+
+  const clearTitle = () => {
+    setTitle('');
+  };
+
+  const clearSelectedUserId = () => {
+    setSelectedUserId(0);
+  };
+
+  const isValidTitle = (text: string) => {
+    const pattern = /[$&+,:;=?@#|№<>."_^*()%!]/g;
+
+    return !pattern.test(text);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.currentTarget.value);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    clearButtonClick();
+  };
+
+  const onSubmit = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setIsButtonClicked(true);
+
+    if (title.trim() && selectedUserId && isValidTitle(title)) {
+      addTodo();
+      clearTitle();
+      clearSelectedUserId();
+      clearButtonClick();
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="createTitle">
+            Tilte:&nbsp;
+          </label>
+
+          <input
+            id="createTitle"
+            type="text"
+            value={title}
+            placeholder="Enter a title"
+            data-cy="titleInput"
+            onChange={handleInputChange}
+          />
+
+          {isButtonClicked
+          && !title
```

Comment:
you should create separate boolean variables in state for handling errors for each input